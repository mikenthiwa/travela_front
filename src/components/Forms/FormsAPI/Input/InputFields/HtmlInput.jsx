import React from 'react';
import {PropTypes} from 'prop-types';

// good for types like text, email, passord 
const HtmlInput = ({...props, value = ''}) => {
  let htmlProps = {...props, value};
  let { error } = props;
  let className = error? 'error': '';
  ['labelNote', 'toggleOptions', 'activeOption', 'selectedDate', 'error']
    .map(i => delete htmlProps[i]); // onChange has been included in htmlProps

  return (
    <input
      {...htmlProps}
      className={`${className} ${htmlProps.className}`}
    />
  );
};

HtmlInput.propTypes = {
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

HtmlInput.defaultProps = {
  error: ''
};

HtmlInput.propTypes = {
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool
  ])
};

export default HtmlInput;
