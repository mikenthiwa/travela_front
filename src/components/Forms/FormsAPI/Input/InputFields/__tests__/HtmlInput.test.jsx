import React from 'react';
import HtmlInput from '../HtmlInput';

describe('<HtmlInput />', () => {

  it('renders as expected', () => {
    const wrapper = shallow(<HtmlInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows receives error class name', () => {
    const wrapper = shallow(<HtmlInput error="This field is required" />);
    expect(wrapper.prop('className')).toContain('error');
  });

});
