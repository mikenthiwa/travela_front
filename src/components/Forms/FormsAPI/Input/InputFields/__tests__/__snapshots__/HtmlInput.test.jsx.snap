// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlInput /> renders as expected 1`] = `
<input
  className=" undefined"
  value=""
/>
`;
